services/QuillLMS/engines/comprehension/app/models/comprehension/rule.rb

```diff
@@ -80,6 +82,63 @@ def display_name
       DISPLAY_NAMES[rule_type.to_sym] || rule_type
     end
 
+    def plagiarism?
+      rule_type == TYPE_PLAGIARISM
+    end
+
+    def regex?
+      rule_type == TYPE_REGEX_ONE || rule_type == TYPE_REGEX_TWO || rule_type == TYPE_REGEX_THREE
+    end
+
+    def automl?
+      rule_type == TYPE_AUTOML
+    end
+
+    def universal_rule_type?
+      rule_type == TYPE_SPELLING || rule_type == TYPE_GRAMMAR
+    end
+
+    def universal?
+      universal
+    end
+
+    def change_log_name
```

Comment:
This you can take or leave since your approach works, but similar to the banner project and the use of `DISPLAY_NAMES` above, you could create a map of `rule_type` to `name` and just do a hash lookup here, e.g.

```ruby
NAMES_FOR_TYPES = {
  TYPE_AUTOML => "Semantic Label",
  TYPE_PLAGIARISM  => "Plagiarism Rule"
...
}

def change_log_name
  NAMES_FOR_TYPES[rule_type] || ""
end